lib/ansible/plugins/filter/core.py

```diff
@@ -132,7 +132,11 @@ def regex_replace(value='', pattern='', replacement='', ignorecase=False, multil
     if multiline:
         flags |= re.M
     _re = re.compile(pattern, flags=flags)
-    return _re.sub(replacement, value)
+    (output, subs) = _re.subn(replacement, value, count=count)
+    if mandatory_count not in {0, subs}:
```

Comment:
`if mandatory_count and mandatory_count == subs`
Just in case someone sets it to `None/null`